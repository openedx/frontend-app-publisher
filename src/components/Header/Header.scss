.site-header {
  .navbar-brand {
    max-width: 100px;
  }

  .disabled-link {
    pointer-events: none;
  }
}
