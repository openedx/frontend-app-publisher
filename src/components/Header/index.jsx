import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import qs from 'query-string';
import { Link } from 'react-router-dom';
import { redirectToLogout } from '@edx/frontend-platform/auth';
import { AppContext } from '@edx/frontend-platform/react';
import { Dropdown, DropdownButton, Hyperlink } from '@edx/paragon';

import EdxLogo from '../../../assets/edx-sm.png';

const Header = ({ darkModeOn, location, toggleDarkMode }) => {
  const { authenticatedUser } = useContext(AppContext);

  if (darkModeOn) {
    document.body.classList.add('dark-mode');
  } else {
    document.body.classList.remove('dark-mode');
  }

  // Allow users to toggle dark mode on if they put in a querystring like ?bananas=1
  const querystringParams = qs.parse(location.search);
  const allowDarkModeToggle = querystringParams.bananas;

  return (
    <header className="site-header mb-3 py-3 border-bottom-blue">
      <div className="container">
        <div className="row align-items-center">
          <div className="col-auto justify-content-start">
            <Hyperlink destination={process.env.STUDIO_BASE_URL}>
              <img src={EdxLogo} alt="edX logo" height="30" width="60" />
            </Hyperlink>
          </div>
          <div className="col">
            <Link to="/">Courses</Link>
          </div>
          {allowDarkModeToggle
            && (
            <div className="col-auto justify-content-end">
              <button type="button" className="btn btn-primary" onClick={toggleDarkMode}>
                Switch to {darkModeOn ? 'light mode' : 'dark mode'}
              </button>
            </div>
            )}
          <div className="col-auto justify-content-end">
            <DropdownButton
              alignRight
              title={authenticatedUser.username}
              variant="light"
            >
              <Dropdown.Item
                onClick={() => redirectToLogout(process.env.STUDIO_BASE_URL)}
              >
                Sign Out
              </Dropdown.Item>
            </DropdownButton>
          </div>
        </div>
      </div>
    </header>
  );
};

Header.propTypes = {
  darkModeOn: PropTypes.bool,
  location: PropTypes.shape({
    search: PropTypes.string,
  }),
  toggleDarkMode: PropTypes.func,
};

Header.defaultProps = {
  darkModeOn: false,
  location: {},
  toggleDarkMode: () => {},
};

export default Header;
