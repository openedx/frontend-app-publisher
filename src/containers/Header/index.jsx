import { connect } from 'react-redux';
import { withRouter } from 'react-router';

import Header from '../../components/Header';
import toggleDarkMode from '../../data/actions/darkMode';

const mapStateToProps = state => ({
  darkModeOn: state.darkMode.darkModeOn,
});

const mapDispatchToProps = {
  toggleDarkMode,
};

const HeaderContainer = connect(mapStateToProps, mapDispatchToProps)(Header);
export default withRouter(HeaderContainer);
