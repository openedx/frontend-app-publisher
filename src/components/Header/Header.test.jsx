import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { AppContext } from '@edx/frontend-platform/react';

import Header from './index';

describe('Header', () => {
  const HeaderWrapper = props => (
    mount(
      <AppContext.Provider
        value={{
          authenticatedUser: {
            username: 'test-username',
          },
        }}
      >
        <MemoryRouter>
          <Header {...props} />
        </MemoryRouter>
      </AppContext.Provider>,
    ).find('header')
  );

  it('renders the correct header', () => {
    const component = HeaderWrapper({});
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('renders the header correctly when users pass a querystring param to allow dark mode toggling', () => {
    const component = HeaderWrapper({ location: { search: '?bananas=1' } });
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('renders the header correctly when toggling is allowed, and dark mode is on', () => {
    const component = HeaderWrapper({ darkModeOn: true, location: { search: '?bananas=1' } });
    expect(shallowToJson(component)).toMatchSnapshot();
  });
});
