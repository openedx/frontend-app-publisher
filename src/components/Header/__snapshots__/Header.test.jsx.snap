// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders the correct header 1`] = `
<header
  className="site-header mb-3 py-3 border-bottom-blue"
>
  <div
    className="container"
  >
    <div
      className="row align-items-center"
    >
      <div
        className="col-auto justify-content-start"
      >
        <withDeprecatedProps(Hyperlink)
          destination="http://localhost:18010"
        >
          <Hyperlink
            destination="http://localhost:18010"
            externalLinkAlternativeText="Opens in a new window"
            externalLinkTitle="Opens in a new window"
            onClick={[Function]}
            target="_self"
          >
            <a
              href="http://localhost:18010"
              onClick={[Function]}
              target="_self"
            >
              <img
                alt="edX logo"
                height="30"
                src="test-file-stub"
                width="60"
              />
            </a>
          </Hyperlink>
        </withDeprecatedProps(Hyperlink)>
      </div>
      <div
        className="col"
      >
        <Link
          to="/"
        >
          <LinkAnchor
            href="/"
            navigate={[Function]}
          >
            <a
              href="/"
              onClick={[Function]}
            >
              Courses
            </a>
          </LinkAnchor>
        </Link>
      </div>
      <div
        className="col-auto justify-content-end"
      >
        <DropdownButton
          alignRight={true}
          title="test-username"
          variant="light"
        >
          <Dropdown
            alignRight={true}
            navbar={false}
          >
            <ReactOverlaysDropdown
              alignEnd={true}
              itemSelector=".dropdown-item:not(.disabled):not(:disabled)"
              onToggle={[Function]}
            >
              <div
                className="dropdown"
                onKeyDown={[Function]}
              >
                <DropdownToggle
                  variant="light"
                >
                  <Button
                    active={false}
                    aria-expanded={false}
                    aria-haspopup={true}
                    className="dropdown-toggle"
                    disabled={false}
                    onClick={[Function]}
                    variant="light"
                  >
                    <button
                      aria-expanded={false}
                      aria-haspopup={true}
                      className="dropdown-toggle btn btn-light"
                      disabled={false}
                      onClick={[Function]}
                      type="button"
                    >
                      test-username
                    </button>
                  </Button>
                </DropdownToggle>
                <DropdownMenu
                  alignRight={false}
                  flip={true}
                />
              </div>
            </ReactOverlaysDropdown>
          </Dropdown>
        </DropdownButton>
      </div>
    </div>
  </div>
</header>
`;

exports[`Header renders the header correctly when toggling is allowed, and dark mode is on 1`] = `
<header
  className="site-header mb-3 py-3 border-bottom-blue"
>
  <div
    className="container"
  >
    <div
      className="row align-items-center"
    >
      <div
        className="col-auto justify-content-start"
      >
        <withDeprecatedProps(Hyperlink)
          destination="http://localhost:18010"
        >
          <Hyperlink
            destination="http://localhost:18010"
            externalLinkAlternativeText="Opens in a new window"
            externalLinkTitle="Opens in a new window"
            onClick={[Function]}
            target="_self"
          >
            <a
              href="http://localhost:18010"
              onClick={[Function]}
              target="_self"
            >
              <img
                alt="edX logo"
                height="30"
                src="test-file-stub"
                width="60"
              />
            </a>
          </Hyperlink>
        </withDeprecatedProps(Hyperlink)>
      </div>
      <div
        className="col"
      >
        <Link
          to="/"
        >
          <LinkAnchor
            href="/"
            navigate={[Function]}
          >
            <a
              href="/"
              onClick={[Function]}
            >
              Courses
            </a>
          </LinkAnchor>
        </Link>
      </div>
      <div
        className="col-auto justify-content-end"
      >
        <button
          className="btn btn-primary"
          onClick={[Function]}
          type="button"
        >
          Switch to 
          light mode
        </button>
      </div>
      <div
        className="col-auto justify-content-end"
      >
        <DropdownButton
          alignRight={true}
          title="test-username"
          variant="light"
        >
          <Dropdown
            alignRight={true}
            navbar={false}
          >
            <ReactOverlaysDropdown
              alignEnd={true}
              itemSelector=".dropdown-item:not(.disabled):not(:disabled)"
              onToggle={[Function]}
            >
              <div
                className="dropdown"
                onKeyDown={[Function]}
              >
                <DropdownToggle
                  variant="light"
                >
                  <Button
                    active={false}
                    aria-expanded={false}
                    aria-haspopup={true}
                    className="dropdown-toggle"
                    disabled={false}
                    onClick={[Function]}
                    variant="light"
                  >
                    <button
                      aria-expanded={false}
                      aria-haspopup={true}
                      className="dropdown-toggle btn btn-light"
                      disabled={false}
                      onClick={[Function]}
                      type="button"
                    >
                      test-username
                    </button>
                  </Button>
                </DropdownToggle>
                <DropdownMenu
                  alignRight={false}
                  flip={true}
                />
              </div>
            </ReactOverlaysDropdown>
          </Dropdown>
        </DropdownButton>
      </div>
    </div>
  </div>
</header>
`;

exports[`Header renders the header correctly when users pass a querystring param to allow dark mode toggling 1`] = `
<header
  className="site-header mb-3 py-3 border-bottom-blue"
>
  <div
    className="container"
  >
    <div
      className="row align-items-center"
    >
      <div
        className="col-auto justify-content-start"
      >
        <withDeprecatedProps(Hyperlink)
          destination="http://localhost:18010"
        >
          <Hyperlink
            destination="http://localhost:18010"
            externalLinkAlternativeText="Opens in a new window"
            externalLinkTitle="Opens in a new window"
            onClick={[Function]}
            target="_self"
          >
            <a
              href="http://localhost:18010"
              onClick={[Function]}
              target="_self"
            >
              <img
                alt="edX logo"
                height="30"
                src="test-file-stub"
                width="60"
              />
            </a>
          </Hyperlink>
        </withDeprecatedProps(Hyperlink)>
      </div>
      <div
        className="col"
      >
        <Link
          to="/"
        >
          <LinkAnchor
            href="/"
            navigate={[Function]}
          >
            <a
              href="/"
              onClick={[Function]}
            >
              Courses
            </a>
          </LinkAnchor>
        </Link>
      </div>
      <div
        className="col-auto justify-content-end"
      >
        <button
          className="btn btn-primary"
          onClick={[Function]}
          type="button"
        >
          Switch to 
          dark mode
        </button>
      </div>
      <div
        className="col-auto justify-content-end"
      >
        <DropdownButton
          alignRight={true}
          title="test-username"
          variant="light"
        >
          <Dropdown
            alignRight={true}
            navbar={false}
          >
            <ReactOverlaysDropdown
              alignEnd={true}
              itemSelector=".dropdown-item:not(.disabled):not(:disabled)"
              onToggle={[Function]}
            >
              <div
                className="dropdown"
                onKeyDown={[Function]}
              >
                <DropdownToggle
                  variant="light"
                >
                  <Button
                    active={false}
                    aria-expanded={false}
                    aria-haspopup={true}
                    className="dropdown-toggle"
                    disabled={false}
                    onClick={[Function]}
                    variant="light"
                  >
                    <button
                      aria-expanded={false}
                      aria-haspopup={true}
                      className="dropdown-toggle btn btn-light"
                      disabled={false}
                      onClick={[Function]}
                      type="button"
                    >
                      test-username
                    </button>
                  </Button>
                </DropdownToggle>
                <DropdownMenu
                  alignRight={false}
                  flip={true}
                />
              </div>
            </ReactOverlaysDropdown>
          </Dropdown>
        </DropdownButton>
      </div>
    </div>
  </div>
</header>
`;
